### ISSUE TITLE
Intermittent UI test failure - org.mozilla.fenix.ui.SmokeTest - createFirstCollectionTest

### ISSUE THREAD

**Issue by Archaeopteryx** [mozilla-mobile/fenix#20053]
Failed 2/2 runs.

### Firebase Test Run:
https://console.firebase.google.com/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/7874341191064920003/executions/bs.10c5964153bce0d2/testcases/1/test-cases
### Stacktrace:
```
java.lang.AssertionError
at org.junit.Assert.fail(Assert.java:87)
at org.junit.Assert.assertTrue(Assert.java:42)
at org.junit.Assert.assertNotNull(Assert.java:713)
at org.junit.Assert.assertNotNull(Assert.java:723)
at org.mozilla.fenix.helpers.ext.WaitNotNullKt.waitNotNull(WaitNotNull.kt:20)
at org.mozilla.fenix.ui.robots.TabDrawerRobot.verifySnackBarText(TabDrawerRobot.kt:138)
at org.mozilla.fenix.ui.SmokeTest$createFirstCollectionTest$9.invoke(SmokeTest.kt:903)
at org.mozilla.fenix.ui.SmokeTest$createFirstCollectionTest$9.invoke(SmokeTest.kt:58)
at org.mozilla.fenix.ui.robots.TabDrawerRobotKt.tabDrawer(TabDrawerRobot.kt:369)
at org.mozilla.fenix.ui.SmokeTest.createFirstCollectionTest(SmokeTest.kt:902)
at java.lang.reflect.Method.invoke(Native Method)
at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
at androidx.test.internal.runner.junit4.statement.RunBefores.evaluate(RunBefores.java:80)
at androidx.test.internal.runner.junit4.statement.RunAfters.evaluate(RunAfters.java:61)
at androidx.test.rule.ActivityTestRule$ActivityStatement.evaluate(ActivityTestRule.java:531)
at androidx.test.rule.ActivityTestRule$ActivityStatement.evaluate(ActivityTestRule.java:531)
at androidx.test.rule.GrantPermissionRule$RequestPermissionStatement.evaluate(GrantPermissionRule.java:134)
at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
at org.junit.runners.Suite.runChild(Suite.java:128)
at org.junit.runners.Suite.runChild(Suite.java:27)
at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
at androidx.test.internal.runner.TestExecutor.execute(TestExecutor.java:56)
at androidx.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:395)
at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:2145)
```
### Build: 
https://github.com/mozilla-mobile/fenix/commit/f198e110d7c0eabff455a028e7739bcb6f1abf3b

**Comment by jonalmeida** [comment]
We've backed out the offending changes.

@Archaeopteryx feel free to cc the PR author or reviewer, it may result in a quicker turn around time. I wasn't aware of this failure until Christian noticed the AC build failure and pinged me (since I reviewed the PR the Codrut's day has ended). 🙂

**Comment by jonalmeida** [comment]
With https://github.com/mozilla-mobile/fenix/pull/20055 backed out, we're still seeing the failures, but less intermittently.

It looks like the [test was re-enabled](https://github.com/mozilla-mobile/fenix/commit/6278a2b07d35c87d5d6d138573d33a99de9a10dc#diff-37b3260fc530d5fb971d35603b1285cd7fd2e656aac9c9f0c6951f65ca6a995d) after a patch landed for https://github.com/mozilla-mobile/fenix/issues/19090, but that patch did not fix the issue [entirely](https://github.com/mozilla-mobile/fenix/issues/19090#issuecomment-861442775).

**Comment by jonalmeida** [comment]
Hmm.. https://github.com/mozilla-mobile/fenix/pull/20061 didn't have the revert in the commit log here, but I still think I've done the right thing by disabling the UI tests because #19739  does not have any thing to do with the test failures and if anything, fixes a regression that was not caught before.

I'm going to try and re-land the fixes for #19739 .

**Comment by jonalmeida** [comment]
The tests are re-enabled and Roger found the cause for the intermittent failures. Let's close this for now unless it re-appears.
